src/hooks/useUpdateObjectRelationships.tsx

```diff
@@ -0,0 +1,60 @@
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { RequestDocument } from "graphql-request";
+
+import {
+  GQLSkylarkUpdateObjectContentResponse,
+  ParsedSkylarkObjectRelationships,
+  SkylarkObjectType,
+} from "src/interfaces/skylark";
+import { skylarkRequest } from "src/lib/graphql/skylark/client";
+import { createUpdateObjectRelationshipsMutation } from "src/lib/graphql/skylark/dynamicMutations";
+
+import { createGetObjectRelationshipsKeyPrefix } from "./useGetObjectRelationships";
+import { useSkylarkObjectOperations } from "./useSkylarkObjectTypes";
+
+export const useUpdateObjectRelationships = ({
+  objectType,
+  uid,
+  updatedRelationshipObjects,
+  originalRelationshipObjects,
+  onSuccess,
+}: {
+  objectType: SkylarkObjectType;
+  uid: string;
+  updatedRelationshipObjects: ParsedSkylarkObjectRelationships[] | null;
+  originalRelationshipObjects: ParsedSkylarkObjectRelationships[] | null;
+  onSuccess: () => void;
+}) => {
+  const queryClient = useQueryClient();
+  const { objectOperations } = useSkylarkObjectOperations(objectType);
+
+  const updateObjectRelationshipsMutation =
+    createUpdateObjectRelationshipsMutation(
+      objectOperations,
+      updatedRelationshipObjects,
+      originalRelationshipObjects,
+    );
+
+  const { mutate, ...rest } = useMutation({
+    mutationFn: ({ uid }: { uid: string }) => {
+      return skylarkRequest<GQLSkylarkUpdateObjectContentResponse>(
```

Comment:
`GQLSkylarkUpdateObjectContentResponse`
This should be specific to the relationship response